lib/src/idx/planner/plan.rs

```diff
@@ -2,22 +2,30 @@ use crate::err::Error;
 use crate::idx::ft::MatchRef;
 use crate::idx::planner::tree::Node;
 use crate::sql::statements::DefineIndexStatement;
+use crate::sql::with::With;
 use crate::sql::Object;
 use crate::sql::{Expression, Idiom, Operator, Value};
 use std::collections::HashMap;
 use std::hash::Hash;
 use std::sync::Arc;
 
-pub(super) struct PlanBuilder {
+pub(super) struct PlanBuilder<'a> {
 	indexes: Vec<(Expression, IndexOption)>,
+	with: &'a Option<With>,
 	all_and: bool,
 	all_exp_with_index: bool,
 }
 
-impl PlanBuilder {
-	pub(super) fn build(root: Node) -> Result<Plan, Error> {
+impl<'a> PlanBuilder<'a> {
+	pub(super) fn build(root: Node, with: &'a Option<With>) -> Result<Plan, Error> {
+		if let Some(with) = with {
+			if matches!(with, With::NoIndex) {
+				return Err(Error::BypassQueryPlanner);
```

Comment:
Is this really an error that users should handle? Shouldn't the return type be `Result<Option<Plan>, Error>` and then `return Ok(None);` here?

Reply:
`Error::BypassQueryPlanner` is never shown to the user. It is caught higher in the code to fall back to the Table Iterator. I use it in different places in the Query Planner

https://github.com/surrealdb/surrealdb/blob/759b489ab91c9fea7cd21ba008444749ec0851e0/lib/src/idx/planner/mod.rs#L62

Reply:
That's what I thought @emmanuel-keller. IMHO this is not an error. We are returning an optional value.

Reply:
Fixed! (By Error::BypassQUeryPlanner, welcome Plan::TableIterator)

Reply:
Awesome! I love this change @emmanuel-keller! ❤️